jepsen/src/jepsen/control/util.clj

```diff
@@ -104,20 +118,25 @@
 
   Options:
 
-    :force?     Even if we have this cached, download the tarball again anyway."
+    force?      Even if we have this cached, download the tarball again anyway.
+    user        user for wget authentication. If provided, valid pw must also be provided.
+    pw          password for wget authentication."
   ([url]
-   (wget! url {:force? false}))
+   (cached-wget! url {:force? false}))
   ([url opts]
    (let [encoded-url (String. (b64/encode (.getBytes url)) "UTF-8")
-         dest-file   (str wget-cache-dir "/" encoded-url)]
+         dest-file   (str wget-cache-dir "/" encoded-url)
+         wget-opts (concat std-wget-opts [:-O dest-file])]
      (when (:force? opts)
        (info "Clearing cached copy of" url)
        (exec :rm :-rf dest-file))
+     (when-not (empty? (:user? opts))
+       (concat wget-opts [:--user (:user? opts) :--password (:pw? opts)]))
```

Comment:
What... is this supposed to do exactly? It looks like this is a noop. Maybe worth writing integration tests here?

Reply:
Yeah, I hadn't tested this yet, sorry :( I will try to add some tests here and also run some of the existing tests